php/stdin.php

```diff
@@ -0,0 +1,16 @@
+<?php
+// Script to read stdin line by line, transform, and return it
+
+$i = 1;
+
+while (true) {
```

Comment:
This while loop could be more concise:

```php
while (($user_input = fgets(STDIN))) {
    echo $i++ . ' ' . strtoupper($user_input);
}
```

1. We can use the fact that `fgets` will return `false` if there's nothing left to read
2. I'm not sure how I feel about it, but we can use `$i++` to increment `$i` without having to use another line.

Reply:
Nice! But I don't get why `echo $i++`  works:  is `$i++` evaluated after it is printed? I would have thought that staring with `$i = 1;` the first line would be printed with i=2, not i=1

Reply:
Yep! It’s an old feature inherited from C (or maybe an even earlier language). `i++` is incremented after `i` is evaluated. `++i` is incremented before. 

This feature is present in C, C++, PHP, Javascript, Java, and a bunch more languages. 

It’s mostly used in loops, especially the ones that look like `for (i = 0; i<100; i++)`